File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int> count;
        for(int num:nums){
            count[num]++;
        }
        vector<pair<int,int>>result(count.begin(),count.end());
        sort(result.begin(), result.end(), [](const auto& a, const auto& b) {
        return a.second > b.second; // Ascending order by value
        });
        vector<int> output;
        for(int i=0;i<k;i++){
            output.push_back(result[i].first);
        }
        return output;
        
    }
};

File: Data Structures & Algorithms/special-array-with-x-elements-greater-than-or-equal-x/submission-1.java
class Solution {
    public int specialArray(int[] nums) {
        int count =1;
        Arrays.sort(nums);
        for(int i = nums.length-1; i>=0;i--){
            if(nums[i]>=count){
                count++;
            }
            else if(nums[i]==count-1)
            {
                return -1;
            }
            else
            {  
                break;
            }
        }
        return count==1?-1:count-1;
    }
}